File: SDEIArchitecture/app/src/main/java/com/sdei/sdeiarchitecture/repository/networkoperator/ResponseCodes.java
package com.sdei.sdeiarchitecture.repository.networkoperator;

public class ResponseCodes {
    /** System Level Response Codes Starts Here **/

    /**
     * The Constant Success.
     */
    public static final int Success = 200;

    /**
     * The Constant Success.
     */
    public static final int Accepted = 201;

    /**
     * The Constant AccessToken expired.
     */
    public static final int ACCESS_TOKEN_EXPIRED = 401;

    /**
     * The Constant RefreshToken expired.
     */
    public static final int REFRESH_TOKEN_EXPIRED = 400;

    /**
     * The Constant InvalidUseridPassword
     */
    public static final int BAD_REQUEST = 400;


    // Error Codes
    public static final int REQUEST_CANCEL = -1;
    public static final int RESPONSE_JSON_NOT_VALID = 1;
    public static final int MODEL_TYPE_CAST_EXCEPTION = 2;
    public static final int INTERNET_NOT_AVAILABLE = 3;
    public static final int WRONG_URL = 4;
    public static final int WRONG_METHOD_NAME = 5;
    public static final int URL_CONNECTION_ERROR = 6;
    public static final int UNKNOWN_ERROR = 10;

    public static final int NOT_ALLOWED = 403;


    public static String logErrorMessage(int code) {

        switch (code) {

            case REQUEST_CANCEL:
                return "Request Canceled";

            case INTERNET_NOT_AVAILABLE:
                return "Internet connection is not available. Please check it and try again";

            case WRONG_URL:
                return "You are trying to hit wrong url, Please check it and try again";

            case WRONG_METHOD_NAME:
                return "You are passing wrong method name. i.e POST, GET, DELETE etc";

            case URL_CONNECTION_ERROR:
                return "Connection is not established, Please try again";

            case RESPONSE_JSON_NOT_VALID:
                return "Json you are getting is not valid";

            case MODEL_TYPE_CAST_EXCEPTION:
                return "Server is not working. Please try after some time.";

            case NOT_ALLOWED:
                return "Server is not working. Please try after some time.";

            default:
                return "Unknown error";
        }
    }
}

File: app/src/main/java/android/sdei/com/basicapp/repository/ResponseCodes.kt
package android.sdei.com.basicapp.repository

/**
 * @ResponseCodes class contain all the static values of responses
 */
class ResponseCodes {

    companion object {


    /** System Level Response Codes Starts Here **/

    /** The Constant Success.  */
    val Success = 200

    /** The Constant Success.  */
    val Accepted = 201

    /** The Constant AccessToken expired.
     *
     * */
    val ACCESS_TOKEN_EXPIRED = 401

    /** The Constant RefreshToken expired.  */
    val REFRESH_TOKEN_EXPIRED = 400

    /** The Constant InvalidUseridPassword  */
    val BAD_REQUEST = 400


    // Error Codes
    val REQUEST_CANCEL = -1
    val RESPONSE_JSON_NOT_VALID = 1
    val MODEL_TYPE_CAST_EXCEPTION = 2
    val INTERNET_NOT_AVAILABLE = 3
    val WRONG_URL = 4
    val WRONG_METHOD_NAME = 5
    val URL_CONNECTION_ERROR = 6
    val UNKNOWN_ERROR = 10

    val NOT_ALLOWED = 403


    fun logErrorMessage(code: Int): String {
        var errorMessage = ""

        when (code) {

            REQUEST_CANCEL -> errorMessage = "Request Canceled"

            INTERNET_NOT_AVAILABLE -> errorMessage = "Internet connection is not available. Please check it and try again"

            WRONG_URL -> errorMessage = "You are trying to hit wrong url, Please check it and try again"

            WRONG_METHOD_NAME -> errorMessage = "You are passing wrong method name. i.e POST, GET, DELETE etc"

            URL_CONNECTION_ERROR -> errorMessage = "Connection is not established, Please try again"


            RESPONSE_JSON_NOT_VALID ->

                errorMessage = "Json you are getting is not valid"

            MODEL_TYPE_CAST_EXCEPTION -> errorMessage = "Server is not working. Please try after some time."

            NOT_ALLOWED -> errorMessage = "Server is not working. Please try after some time."

            else -> errorMessage = "Unknown error"
        }
        return errorMessage
    }
    }
}